src/main/java/software/amazon/awssdk/crt/CRT.java

```diff
@@ -152,40 +153,83 @@ public static String getCRuntime(String osIdentifier) {
             return NON_LINUX_RUNTIME_TAG;
         }
 
-        Runtime rt = Runtime.getRuntime();
-        String[] commands = {"ldd", "--version"};
-        try {
-            java.lang.Process proc = rt.exec(commands);
-
-            // the "normal" input stream of the proc is the stdout of the invoked command
-            BufferedReader stdOutput = new BufferedReader(new
-                    InputStreamReader(proc.getInputStream()));
-
-            // sometimes, ldd's output goes to stderr, so capture that too
-            BufferedReader stdError = new BufferedReader(new
-                    InputStreamReader(proc.getErrorStream()));
-
-            String line;
-            StringBuilder outputBuilder = new StringBuilder();
-            while ((line = stdOutput.readLine()) != null) {
-                outputBuilder.append(line);
+        // The system might have both musl and glibc on it:
+        // https://github.com/awslabs/aws-crt-java/issues/659
+
+        // First, check which one java is using.
+        // Run: ldd $JAVA_HOME/bin/java
+        // If musl, has a line like: libc.musl-x86_64.so.1 => /lib/ld-musl-x86_64.so.1 (0x7f7732ae4000)
+        // If glibc, has a line like: libc.so.6 => /lib64/ld-linux-x86-64.so.2 (0x7f112c894000)
+        Pattern muslWord = Pattern.compile("\\bmusl\\b", Pattern.CASE_INSENSITIVE);
+        Pattern libcWord = Pattern.compile("\\blibc\\b", Pattern.CASE_INSENSITIVE);
```

Comment:
There are N+1 Linux distributions out there.
I guarantee someone's working on new distro now just because they wish "musl" was capitalized differently.